rvgo/fast/vm.go

```diff
@@ -656,10 +656,10 @@ func (inst *InstrumentedState) riscvStep() (outErr error) {
 			rdValue = mask32Signed64(shl64(and64(imm, toU64(0x1F)), rs1Value))
 		case 5: // 101 = SR~
 			shamt := and64(imm, toU64(0x1F))
-			switch shr64(toU64(6), imm) { // in rv64i the top 6 bits select the shift type
-			case 0x00: // 000000 = SRLIW
+			switch shr64(toU64(6), imm) { // top 7 bits select the shift type
```

Comment:
```suggestion
			switch shr64(toU64(5), imm) { // top 7 bits select the shift type
```
shouldn't shift amount be 5 to get top 7 bits?

Reply:
Thanks for the catch. Fixed at https://github.com/ethereum-optimism/asterisc/pull/22/commits/ca349af61e9be6f82b4f529932bff734f14509eb